frontend/src/pages/Chat/ChatComponent.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState, useEffect } from "react";
+import OnlineUsersList from "./OnlineUsers";
+import ChatWindow from "./ChatWindow";
+import { getHistoryMessages } from "../../services/auth.service";
+import io, { Socket } from "socket.io-client";
+import useCurrentUser from "../../hooks/useCurrentUser";
+import { User } from "../../interfaces/User";
+
+interface Message {
+  sender: string;
+  receiver: string;
+  text: string;
+}
+
+const ChatComponent = () => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [messages, setMessages] = useState<Record<string, Message[]>>({});
+  const [newMessage, setNewMessage] = useState("");
+  const [socket, setSocket] = useState<Socket | null>(null);
+  const [isSocketConnected, setIsSocketConnected] = useState<boolean | null>(
+    null
+  );
+  const { user: currentUser } = useCurrentUser();
+
+  useEffect(() => {
+    if (!currentUser) {
+      return;
+    }
+
+    const newSocket = io("http://localhost:3000", {
+      query: { userId: String(currentUser?._id) },
+    });
+
+    newSocket.on("connect", () => setIsSocketConnected(true));
+    newSocket.on("disconnect", () => setIsSocketConnected(false));
+
+    const handlePrivateMessageReceived = (receivedMessage: {
+      message: Message;
+    }) => {
+      const message = receivedMessage.message;
+      console.log("Message received", "sender", message.sender, message);
+      setMessages((prevMessages) => {
+        const updatedMessages = { ...prevMessages };
+        if (!updatedMessages[message.sender]) {
+          return updatedMessages;
+        }
+        updatedMessages[message.sender] = [
+          ...updatedMessages[message.sender],
+          message,
+        ];
+        return updatedMessages;
+      });
+    };
+
+    newSocket.on("private_message_received", handlePrivateMessageReceived);
+
+    setSocket(newSocket);
+
+    return () => {
+      newSocket.disconnect();
+    };
+  }, [currentUser]);
+
+  useEffect(() => {
+    console.log(
+      "All users messages dict:",
+      messages,
+      "SelectedUser",
+      selectedUser
+    );
+  }, [messages, selectedUser]);
+
+  const updateSelectedUser = (user: User | null) => {
+    setSelectedUser(user);
+  };
+
+  const startChat = async (user: User) => {
+    if (!user) return;
+
+    const userId = String(user._id);
+    if (!(userId in messages)) {
+      setSelectedUser(user);
+
+      const historyMessages = await getHistoryMessages(
+        String(currentUser?._id),
+        userId
+      );
```

Comment:
maybe try something like:
`        const historyMessages = await axios.get(`${AUTH_HISTORY_MESSAGES}/${Id}`);
`.

and then as usual. use the frontend endpoint in file `constants.ts` to do an **axios** call. to the backend endpoint in file `message.route.ts` that uses the _messageController_.